model/errors/transaction.go

```diff
@@ -0,0 +1,459 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const (
+	// tx validation errors
+	errCodeInvalidTxByteSizeError     = 1
+	errCodeInvalidReferenceBlockError = 2
+	errCodeExpiredTransactionError    = 3
+	errCodeInvalidScriptError         = 4
+	errCodeInvalidGasLimitError       = 5
+	errCodeInvalidAddressError        = 6
+	errCodeInvalidArgumentError       = 7
+
+	// execution errors
+	errCodeProposalMissingSignatureError  = 50
+	errCodeProposalSeqNumberMismatchError = 51
+
+	errCodePayloadSignatureError           = 60
+	errCodePayloadSignatureKeyError        = 61
+	errCodeRevokedPayloadSignatureKeyError = 62
+
+	errCodeEnvelopeSignatureError           = 70
+	errCodeEnvelopeSignatureKeyError        = 71
+	errCodeRevokedEnvelopeSignatureKeyError = 72
+
+	errCodeAuthorizationError = 80
+
+	errCodeCadenceRunTimeError = 100
+
+	errCodeInsufficientStorageError              = 200
+	errCodeInsufficientTokenBalanceError         = 201
+	errCodeMaxGasExceededError                   = 202
+	errCodeMaxEventLimitExceededError            = 203
+	errCodeMaxLedgerIntractionLimitExceededError = 204
+)
+
+// TransactionValidationError captures a transaction validation error
+// A transaction having this error (in most cases) is rejected by access/collection nodes
+// and later in the pipeline be verified by execution and verification nodes.
+type TransactionValidationError interface {
+	// Code returns the code for this error
+	Code() uint32
+	// Error returns an string describing the details of the error
+	Error() string
```

Comment:
I assume the purpose of this is to ensure these can also pass as basic Go errors, in which case I'd suggest we embed the `error` interface here to make that explicit.
```suggestion
	error
```